### PR TITLE
feat(Alert/Banner): Adds ColorModes for backwards compatibility and uses most recent props [GM-224]

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Rebased from previous PRs.

* Refactor Alert and Banner to use Background and switch color modes based on their correct color.
* Updates structure to reflect new system props.
* Adds stories to Alert to confirm support of links in the correct colors.

### PR Checklist

- [x] Related to designs:
- [x] Related to JIRA ticket: GM-224
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.83-alpha.916606.0<br/>@codecademy/gamut-labs@15.5.7-alpha.916606.0<br/>@codecademy/gamut@32.14.1-alpha.916606.0<br/>@codecademy/styleguide@36.16.1-alpha.916606.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609343979404ac1cc77dbb92--gamut-preview.netlify.app>https://609343979404ac1cc77dbb92--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609343979404ac1cc77dbb92>Deploy Logs</a></p>

File: packages/gamut/src/Alert/Alert.tsx [hunk 1]
```diff
@@ -58,155 +39,106 @@ export type InlineAlert = AlertBase & {
...
   'aria-label': 'alert box',
   'aria-live': 'polite',
-};
-
-const AlertContent = styled(motion.div)(
-  ({ theme }) => css`
-    padding: ${theme.spacing[4]} 0;
-    overflow-y: hidden;
-  `
+} as any;
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Alert/Alert.tsx:50]:
```suggestion
};
```

File: packages/gamut/src/Alert/Alert.tsx [hunk 2]
```diff
@@ -58,155 +39,106 @@ export type InlineAlert = AlertBase & {
...
+
+const CollapseButton = styled(IconButton)(
+  system.variant({
+    prop: 'toggleState',
+    defaultVariant: 'collapsed',
+    base: { svg: { transition: '200ms transform' } },
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Alert/Alert.tsx:56]: > `200ms`

We have a `timing` constants file. Let's use it here?

**Review by JoshuaKGoldberg — APPROVED** [review]
🆒

File: packages/gamut/src/Alert/Alert.tsx (hunk 1 above)
Comment (codecaaron) [reply packages/gamut/src/Alert/Alert.tsx:50]: Oh whoops I had a bug cause of Background not fowarding a ref

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.83-alpha.911e5d.0<br/>@codecademy/gamut-labs@15.5.7-alpha.911e5d.0<br/>@codecademy/gamut@32.14.1-alpha.911e5d.0<br/>@codecademy/styleguide@36.16.1-alpha.911e5d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6093ff80273d54093a5872ea--gamut-preview.netlify.app>https://6093ff80273d54093a5872ea--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6093ff80273d54093a5872ea>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.83-alpha.7b4200.0<br/>@codecademy/gamut-labs@15.5.7-alpha.7b4200.0<br/>@codecademy/gamut@32.14.1-alpha.7b4200.0<br/>@codecademy/styleguide@36.16.1-alpha.7b4200.0<br/></pre>

**Review by dreamwasp — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.83-alpha.0e3a5c.0<br/>@codecademy/gamut-labs@15.5.7-alpha.0e3a5c.0<br/>@codecademy/gamut@32.14.1-alpha.0e3a5c.0<br/>@codecademy/styleguide@36.16.1-alpha.0e3a5c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60943311a050bf00d22b3454--gamut-preview.netlify.app>https://60943311a050bf00d22b3454--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60943311a050bf00d22b3454>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.84-alpha.1524f0.0<br/>@codecademy/gamut-labs@15.6.1-alpha.1524f0.0<br/>@codecademy/gamut@32.14.1-alpha.1524f0.0<br/>@codecademy/styleguide@37.0.2-alpha.1524f0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609437701dbae4008a81c89f--gamut-preview.netlify.app>https://609437701dbae4008a81c89f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609437701dbae4008a81c89f>Deploy Logs</a></p>

**Pull request merged**